playground/infrastructure/datastore_client.py

```diff
@@ -83,54 +85,65 @@ def save_to_cloud_datastore(self, examples_from_rep: List[Example], sdk: Sdk, or
         examples_ids_before_updating = self._get_all_examples(sdk, origin)
 
         # loop through every example to save them to the Cloud Datastore
```

Comment:
this huge chunk is only to remove transaction when updating examples: datastore transactions are limited to 500 entities affected, here we have possibly more.
https://cloud.google.com/datastore/docs/concepts/limits

Reply:
The only reasonable restriction is to not leave single example-related entities in inconsistent state
I've refactored this place to reduce transaction scope to a single example

Reply:
SGTM